### DISCUSSION TITLE
Stalwart v0.7.0 needs the port 8080 to be bound.

### DISCUSSION THREAD

**Discussion by AiroPi** [nextcloud/all-in-one#4547] [Questions] [answered]
Not long ago, Stalwart was upgraded to version 0.7.0.
This brought a lot of breaking changes, and yet the Stalwart community-container was updated to this new version without any warning, which is regrettable (Stalwart stopped working after that).

So I'm currently trying to migrate to this new version, trying to make all the right configurations. Nevertheless, this version now offers a web interface, accessible by default on port `8080`. Unfortunately, this port is already in use by the mastercontainer.

Solutions could be:
- to add a rule in the reverse proxy
- to update the container description to allow different ports to be associated with the inside and outside of the container
- to change the configuration of Stalwart to listen to another port than 8080

**Comment by AiroPi** [comment]
@marcoambrosini how did you upgrade to v7.0.x ? With the new version already installed. The migration guide needs the old version (v0.6.0) to be running in order to export the user correctly.

**Comment by szaimen** [comment]
Hi, I am helping @marcoambrosini with this topic.

> * to change the configuration of Stalwart to listen to another port than 8080

This would be the preferred solution from my side.

> The migration guide needs the old version (v0.6.0) to be running in order to export the user correctly.

Do you have some link to this migration guide? Honestly, the v0.7.0 update sounds like a major upgrade (and not minor update), if people need to follow a migration guide, so from my side it is unexpected that v0.6.0 needs to be present but lets see what the guide states.

**Reply by AiroPi** [reply]
https://github.com/stalwartlabs/mail-server/blob/v0.7.0/UPGRADING.md

> Version 0.7.0 of Stalwart Mail Server introduces significant improvements and features that enhance performance and functionality. However, it also comes with multiple breaking changes in the configuration files and a revamped database layout optimized for accessing large mailboxes. Additionally, Stalwart now supports compression for binaries stored in the blob store, further increasing efficiency. **Due to these extensive changes, the recommended approach for upgrading is to perform a clean reinstallation of Stalwart and manually migrate your accounts to the new version.**

The way this migration works is by exporting the account data from the old installation and importing it to the new one.
A solution can be to install v0.6.0 version image, run it by mounting the stalwart data folder, make the export, and import in the aio-stalwart image.

However, I faced some issues when I tried to export. Don't know if it is an issue with my own installation, or if it related to aio or anything. I contacted the dev of stalwart for help.

At the moment, I published my own aio-stalwart image on docker and I'm using it to stay at v0.6.0 until I find a solution to export my accounts without issues.

**Comment by AiroPi** [comment]
Ok I have more informations about migration.
It seems that the current auto-reverse proxy has some problems with stalwart. I will try to find the issue later.
But concerning the migration, here is the steps to follow to **exports** your old accounts:
- stop your reverse proxy & stalwart: `docker stop nextcloud-aio-caddy nextcloud-aio-stalwart`
- start a v0.6.0 stalwart container: `docker run -p 443:443 -v nextcloud_aio_stalwart:/opt/stalwart-mail -d -it stalwartlabs/mail-server:v0.6.0
- enter it: `docker exec -it <CONTAINER_ID> /bin/bash`
- create a directory for your exports: `mkdir /opt/stalwart-mail/exports`
- export your account(s): `stalwart-cli -u https://127.0.0.1/ -c <credentials> export account <account> /opt/stalwart-mail/exports`
- exit and stop this container. Start caddy and stalwart: `docker stop nextcloud-aio-caddy nextcloud-aio-stalwart`
- enter the nextcloud-stalwart container; `docker exec -it nextcloud-aio-stalwart /bin/bash`

Next, you will have to follow the steps at: https://github.com/stalwartlabs/mail-server/blob/v0.7.2/UPGRADING.md
Concerning the port binding, there is currently no official solution.

@szaimen I can open a PR to add a new port binding with informations for changing the default port, but I don't know what port should be used instead of `8080`. Is `8000` fine ?

**Comment by szaimen** [comment, marked as answer]
See https://github.com/nextcloud/all-in-one/pull/4556
